jhove-modules/pdf-hul/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/PdfHeader.java

```
@@ -16,183 +16,141 @@
+    public static final String PDF_1_SIG_HEADER = "%" + PDF_VER1_HEADER_PREFIX; //$NON-NLS-1$
+    public static final String PDF_2_SIG_HEADER = "%" + PDF_VER2_HEADER_PREFIX; //$NON-NLS-1$
+    public static final String POSTSCRIPT_HEADER_PREFIX = "!PS-Adobe-"; //$NON-NLS-1$
+
+    public static final int MAX_VALID_MAJOR_VERSION = 2;
+
+    private final PdfVersion version;
+    private final boolean isPdfACompilant;
+
+    /**
+     * Factory method for {@link PdfHeader} that parses a new instance using the
+     * supplied {@link Parser} instance.
+     *
+     * @param parser
+     *               the {@link Parser} instance that will be used to parse header
+     *               details
+     * @return a new {@link PdfHeader} instance derived using the supplied
+     *         {@link Parser} or <code>null</code> when no header could be found
+     *         and parsed.
+     * @throws PdfException
+     */
+    public static PdfHeader parseHeader(final Parser parser) throws PdfException {
+        Token token = null;
+        String value = null;
+        boolean isPdfACompliant = false;
+        PdfVersion version = null;
+
+        /* Parse file header. */
+        for (;;) {
+            final long offset = parser.getOffset();
+            if (offset > 1024) {
+                throw new PdfMalformedException(
+                        JhoveMessages.getMessageInstance(MessageConstants.PDF_HUL_137,
+                                "Header not found in first 1024 bytes"),
+                        offset); // PDF-HUL-137
+            }
+
+            try {
+                token = parser.getNext(1024L);
+            } catch (final IOException ee) {
+                throw new PdfMalformedException(
+                        JhoveMessages.getMessageInstance(MessageConstants.PDF_HUL_160, ee.getMessage()),
+                        offset); // PDF-HUL-137
+            } catch (PdfException e) {
```

Will this swallow the PDF_HUL_33 and PDF_HUL_34 errors in the parser.getNext?

---

Sorry @samalloing, it will, but I think they're erroneous in this case as they're specifically for handling dictionary/array end characters without encountering a prior start char. We're just looking for header info, so we shouldn't encounter them at all.